### PR TITLE
feat(Logo): Refactor Logo to ts

### PR DESCRIPTION
**Pull request by AadneRo**
- Refactor the Logo component to typescript
- Add deprecated tag to wrongly cased components and unused components
- Add typed docs

### CODE REVIEW

File: packages/dnb-eufemia/src/components/logo/Logo.tsx [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    if (variant === 'compact') {
+      logoType = LogoType.SbankenCompact
+    } else if (variant === 'compactHorizontal') {
+      logoType = LogoType.SbankenHorizontal
+    } else {
+      logoType = LogoType.SbankenDefault
+    }
+  }
+
+  const ratio = defaultRatio[logoType]
```
Comment (AadneRo) [inline packages/dnb-eufemia/src/components/logo/Logo.tsx:143]: I don't know if we still need this, this should be handled by the svg right?

File: packages/dnb-eufemia/src/components/logo/LogoDocs.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { PropertiesTableProps } from '../../shared/types'
+
+export const LogoProperties: PropertiesTableProps = {
```
Comment (AadneRo) [inline packages/dnb-eufemia/src/components/logo/LogoDocs.ts:3]: should we remove all numbers from the types? right now the width and height is just passed down to the svg, I removed the calculations done

File: packages/dnb-eufemia/src/components/logo/Logo.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    width,
+    height,
+    color,
+    alt: altText,
+  }
+
+  validateDOMAttributes(props, rootParams)
+
+  return (
+    <span {...rootParams}>
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/logo/Logo.tsx:172]:
```suggestion
  return (
    <span {...validateDOMAttributes(props, { ...rootParams })}>
```

File: packages/dnb-eufemia/src/shared/locales/en-GB.ts [hunk 4]
```diff
@@ -132,6 +132,7 @@ export default {
     },
     Logo: {
       alt: 'DNB Logo',
+      sbankenAlt: 'Sbanken - a concept from DNB',
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/shared/locales/en-GB.ts:135]:
```suggestion
```

File: packages/dnb-eufemia/src/shared/locales/nb-NO.ts [hunk 5]
```diff
@@ -132,6 +132,7 @@ export default {
     },
     Logo: {
       alt: 'DNB Logo',
+      sbankenAlt: 'Sbanken - et konsept fra DNB logo',
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/shared/locales/nb-NO.ts:135]:
```suggestion
```

**Review by tujoworker — APPROVED** [review]
Looks good 👍

File: packages/dnb-eufemia/src/components/logo/Logo.tsx [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+import { convertSnakeCaseProps } from '../../shared/helpers/withSnakeCaseProps'
+import LogoSvg from './LogoSvg'
+
+export type LogoWidth = number | string
+export type LogoHeight = number | string
+export type LogoVariant = 'default' | 'compact' | 'compactHorizontal'
+
+export type LogoProps = {
+  /**
+   * Define either the width of the logo.
```
Comment (langz) [inline packages/dnb-eufemia/src/components/logo/Logo.tsx:25]:
```suggestion
   * Define the width of the logo.
```

File: packages/dnb-eufemia/src/components/logo/Logo.tsx [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+   */
+  width?: LogoWidth // v11: replace with string
+  /**
+   * Or define the height of the logo.
```
Comment (langz) [inline packages/dnb-eufemia/src/components/logo/Logo.tsx:29]:
```suggestion
   * Define the height of the logo.
```

**Review by langz — APPROVED** [review]
Looks great, I just came up with two minor suggestions/improvements when I read through the PR. Nothing you actually made a change too 😹

**Pull request merged**